src/Microsoft.Health.Dicom.Core/Features/HealthCheck/BackgroundServiceHealthCheck.cs

```diff
@@ -41,13 +41,20 @@ public class BackgroundServiceHealthCheck : IHealthCheck
 
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
-            Task<DateTimeOffset> oldestWaitingToBeDeleated = _backgroundServiceHealthCheckCache.GetOrAddOldestTimeAsync(_indexDataStore.GetOldestDeletedAsync, cancellationToken);
-            Task<int> numReachedMaxedRetry = _backgroundServiceHealthCheckCache.GetOrAddNumExhaustedDeletionAttemptsAsync(
-                t => _indexDataStore.RetrieveNumExhaustedDeletedInstanceAttemptsAsync(_deletedInstanceCleanupConfiguration.MaxRetries, t),
-                cancellationToken);
+            try
```

Comment:
CheckHealthAsync method doesn't seems to be used as of now, still made change to catch any exception in the background service and report health as unhealthy. Thoughts?

Reply:
This health check should be in use, but it is initialized via DI and triggered at the endpoint `/health/check`.

While you should indeed catch exceptions that you know can be thrown, we should use "catchalls" very sparingly. Note that if an `IHealthCheck` throws an exception that it result in a health check failure automatically:

See [here](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-5.0#create-health-checks):
> If CheckHealthAsync throws an exception during the check, a new HealthReportEntry is returned with its HealthReportEntry.Status set to the FailureStatus, which is defined by AddCheck (see the Register health check services section) and includes the inner exception that initially caused the check failure. The Description is set to the exception's message.

Reply:
Thanks Will for the information. I have pushed the commit to catch specific exception instead of generic one.